docs/Contributing/API-for-contributors.md

```diff
@@ -1712,4 +1714,63 @@ Returns the URL to open when clicking the "Transparency" menu item in Fleet Desk
 
 Redirects to the transparency URL.
 
+### Check if an installer exists 
```

Comment:
This is looking awesome! It's exciting that the Interface team has API specs to work off of and, when this PR is merged, and API to hook the frontend up to. 

I left two questions and a piece of feedback below:

1. I don't think the UI will be using the "Check if an installer exists" endpoint. @lukeheath is this correct?

2. @roperzh and @lucasmrod is there an intended use for this endpoint other than the Fleet UI?

If yes to (1) and no to (2) I think we can wait to include this API endpoint. My thinking is this will reduce the amount of time we spend testing Fleet Sandbox and maybe lead to a sooner release.

Reply:
I know Lucas and @lukeheath had a brief chat about this in the original issue https://github.com/fleetdm/fleet/issues/6365

IMO, checking in `config.license` to show the link is not enough, we might not have the installer ready for whatever reason, which will lead to "user clicks link, link doesn't work"

If we instead use this endpoint, we can be certain that there's an installer and show the link.

Reply:
@lukeheath I'm in for whatever you folks decide, as my UX skills are very poor, but my 2 cents:

Why can't we show the UI with instructions to build your own installer if an installer is not present for whatever reason? kind of:

```
if (isDemo && await checkInstaller()) {
     return <ModalWithDownloadLink />
}

return <ModalWithBuildInstructions />
```

Reply:
@roperzh Showing the modal with the command-line instructions is a good fallback option. However, the goal of sandbox is to make the installer process single click, so we would be failing to achieve our goal. I think the question for @noahtalerman and I is why is it necessary to have a fallback option if we control the end-to-end process? I would consider this type of fallback option necessary if we were relying on a third party and could not guarantee the packages are present. However, since we control the end-to-end process, it seems like we should be able to guarantee they will be there.

Ultimately this will be @noahtalerman's call, so just my two cents as well.

Reply:
@roperzh 

> t's just that in cloud services guaranteeing a very high availability is no easy task, requires a lot of resources, and I don't expect us to get it right from the first version!

That makes sense. If the platform team feels there is a chance that the installers may not be generated properly in all cases it sounds like we do need to have a contingency plan. In that case, I think your suggestion of falling back to the default add host modal is the best option.

@noahtalerman What do you think?

Reply:
I might be overly conservative though, probably worth looping @chiiph

Reply:
Am ok with not having the `HEAD` option in this first iteration. 

> I understand your point, it's just that in cloud services, guaranteeing a very high availability is no easy task, requires a lot of resources, and I don't expect us to get it right from the first version!

We'll need to add proper logging to alert us when this happens.
Good news is that Fleet (we) control sandboxes, so we should know when availability issues happen and fix them.

Reply:
To over-communicate, we have removed the HEAD endpoint 🗣️

Reply:
IMO, if something can fail, it _will_ fail, and we should prepare for it. Having the instructions as a fallback doesn't sound too complicated and it's still close to the experience people have when using Fleet. It just happens to not be the one we are shooting for.

To me, having to scramble to make sure some installers are generated doesn't sound like a better approach than showing instructions to the user and having the right alarms on the infra side to fix the issue asap.

I would rather have the feedback "Sandbox could be better because it should provide installer download links" rather than "Sandbox is unusable because I have no idea how to connect a host to it".

Just my 2c as I was tagged. I'm happy with whatever the majority picks.

Reply:
Thanks for the feedback, @chiiph. 

Implementing the `HEAD` check and showing fallback instructions is definitely not hard to implement on the frontend, I think it's just a UX concern. The alternative is that we show an error message if the files are not there, which is how it's currently designed. But I agree that falling back to the default flow is better UX than failing altogether.

Reply:
So with the above conversation in mind, I'm in favor of the frontend using the HEAD checks to ensure the files are present, and falling back to the default command line instructions if not. Just need to get the 👍  from @noahtalerman and I'll update the frontend specs accordingly.

Reply:
It occurred to me during the weekend that even if @noahtalerman decides to show the command line instructions on error, you still don't need the `HEAD` endpoints, as the `GET` endpoint returns a`404` if the installer is not present. This way you don't have to do 8 HEAD requests!

Reply:
+1 to Roberto's idea. Plus if we offer both a `HEAD` and `GET`, and the `HEAD` works but the `GET` does not (for any reason) then we end up in the same place (user cannot enroll hosts to Sandbox).

Also, another idea: always show an "Advanced" section (that shows the current screen with `fleetctl` commands). That way if there are any issues downloading the installers the user has a fallback option.

Reply:
> To over-communicate, we have removed the HEAD endpoint 🗣️

Sounds good👍 Thanks for the over-communication here!

> the installers not being present would break the sandbox experience, so shouldn't be considered an acceptable outcome.

I agree with Luke's thoughts above. 

I don't think we should display the `fleetctl package` command if the installer isn't ready. In most cases, when the user first enters Fleet Sandbox, the installers should be ready for download. If not, the user will be presented with an error message.

This is because, for Fleet Sandbox, the goal is to provide a demo environment of Fleet, in which the user can see information about their host, without any `fleetctl` commands.

If Fleet Sandbox were to display `fleetctl package` command, a user without experience, or patience for, CLI tools wouldn't be able to successfully see information about their host.

> if something can fail, it will fail

I also agree with Tomas's thoughts here. 

I think it makes sense to have some way to be notified when a Fleet Sandbox instance fails to have packages available for download (ex. Slack message sent to an internal channel).

This way, we can make the user who's experiencing issues feel heard (email) and provide them with resources to be resolve the issue and be successful demoing Fleet.